File: UELPMAPI/UELPM.Model/OpportunityProductsDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace UELPM.Model
{
    public class OpportunityProductsDTO
    {
        public int OpportunityProductId { get; set; }

        public int? OpportunityId { get; set; }

        public int? OpportunityProductType { get; set; }

        public string WritePName { get; set; }

        public string WritePAlias { get; set; }

        public string WritePDesc { get; set; }

        public int? ProductId { get; set; }

        public int? UnitsId { get; set; }

        public int? PricingType { get; set; }

        public decimal? PricePerUnit { get; set; }

        public decimal? VolDiscount { get; set; }

        public int? Qty { get; set; }

        public decimal? Amount { get; set; }

        public decimal? ManualDiscount { get; set; }

        public decimal? Tax { get; set; }

        public decimal? ExtendedAmount { get; set; }

        public DateTime? RequestDate { get; set; }

        public decimal? Price { get; set; }

        public decimal? DiscPer { get; set; }

        public int? PriceListId { get; set; }

        public bool? HasInspection { get; set; }

        public bool? HasCostSheet { get; set; }

        public decimal? CostSheetValue { get; set; }

        public int? CreatedBy { get; set; }

        public DateTime? CreatedDate { get; set; }

        public int? UpdatedBy { get; set; }

        public DateTime? UpdatedDate { get; set; }

        public int? DeletedBy { get; set; }

        public DateTime? DeletedDate { get; set; }

    }
    public class OpportunityProductExist
    {
        public int OpportunityId { get; set; }

        public string Status { get; set; }

        public string Message { get; set; }

        public int ProductId { get; set; }
    }
    public class OpportunityProductResults
    {
        public int OpportunityId { get; set; }

        public IEnumerable<OpportunityProductExist> OpportunityProductExists { get; set; }

        public string Status { get; set; }

        public string Message { get; set; }

        public string StatusCode { get; set; }

        public string Error { get; set; }

        public bool IsProductExists { get; set; }

        public int TotalProductsExists { get; set; }
    }


    public class OpportunityProductsInput
    {
        public int OpportunityId { get; set; }
        public int CreatedBy { get; set; }

        public List<OpportunityProductsDTO> opportunityProductsDTOs { get; set; }
    }

}

File: UELPMAPP/src/app/crm/models/OpportunityProductsDTO.ts
export class  OpportunityProductsDTO {
    opportunityProductId: number;
    opportunityId: number;
    opportunityProductType: number;
    writePName: string;
    writePAlias: string;
    writePDesc: string;
    productId: number;
    unitsId: number;
    pricingType: number;
    pricePerUnit: number;
    volDiscount: number;
    qty: number;
    amount: number;
    manualDiscount: number;
    tax: number;
    extendedAmount: number;
    requestDate: string;
    price: number;
    discPer: number;
    priceListId: number;
    hasInspection: boolean;
    hasCostSheet: boolean;
    costSheetValue: number;
    createdBy: number;
    createdDate: string;
    updatedBy: number;
    updatedDate: string;
    deletedBy: number;
    deletedDate: string;
}
// POPDistributionSummaryItems: Array<POPDistributionSummary>;

export class OpportunityProductsInput {
    opportunityId: number;
    createdBy: number;
    opportunityProductsDTOs:Array<OpportunityProductsDTO>=[];

    // constructor()
    // {

    // }
}

export interface OpportunityProductsResult {
    OpportunityProductsLists: OpportunityProductsList[];
    TotalRecords: number;
    Status: string;
    Message: string;
  }

export interface OpportunityProductsList {
    OpportunityProductId: number;
    OpportunityId: number;
    OpportunityProductType: number;
    ProductId: number;
    ProductCode: string;
    ProductSerialNo: string;
    ProductName: string;
    ProductFamilyId: number;
    ProductDescription: string;
    ProductIsActive: boolean;
    CategoryID: number;
    CreatedBy: number;
    CreatedDate: string;
    UnitsId: number;
    PricingType: number;
    PricePerUnit: number;
    VolDiscount: number;
    Qty: number;
    Amount: number;
    ManualDiscount: number;
    Tax: number;
    ExtendedAmount: number;
    RequestDate: string;
    Price: number;
    DiscPer: number;
    PriceListId: number;
  }

